lib/lockbox/model.rb

```diff
@@ -137,21 +137,22 @@ def attributes
                 # essentially a no-op if already loaded
                 # an exception is thrown if decryption fails
                 self.class.lockbox_attributes.each do |_, lockbox_attribute|
-                  # don't try to decrypt if no decryption key given
-                  next if lockbox_attribute[:algorithm] == "hybrid" && lockbox_attribute[:decryption_key].nil?
-
                   # it is possible that the encrypted attribute is not loaded, eg.
                   # if the record was fetched partially (`User.select(:id).first`).
                   # accessing a not loaded attribute raises an `ActiveModel::MissingAttributeError`.
-                  encrypted_attribute = lockbox_attribute[:encrypted_attribute]
-                  if has_attribute?(encrypted_attribute)
-                    send(lockbox_attribute[:attribute])
+                  if has_attribute?(lockbox_attribute[:encrypted_attribute])
+                    begin
+                      send(lockbox_attribute[:attribute])
+                    rescue ArgumentError => e
+                      raise e if e.message != "No decryption key set"
+                    end
```

Comment:
This was added in the upstream, so I decided to keep it. Same logic, with the added bonus of friendlier error message.